nwb_conversion_tools/utils/si013nwbephyswriter.py

```diff
@@ -0,0 +1,302 @@
+
+    def _get_unit_spike_train_ids(self, unit_id, start_frame=None, end_frame=None, segment_index=None):
+        return self.sorting.get_unit_spike_train(unit_id, start_frame=start_frame, end_frame=end_frame)
+
+    def _get_unit_spike_train_times(self, unit_id, segment_index=0):
+        return self.sorting.frame_to_time(self.sorting.get_unit_spike_train(unit_id=unit_id))
+
+    def _get_unit_property_names(self):
+        property_names = set()
+        for unit_id in self._get_unit_ids():
+            for i in self.sorting.get_unit_property_names(unit_id):
+                property_names.add(i)
+        return list(property_names)
+
+    def _get_unit_property_values(self, prop):
+        prop_values = self._fill_missing_property_values(self._get_unit_ids(), prop, self.sorting.get_unit_property)
+        return self._check_valid_property(prop_values)
+
+    def _get_unit_waveforms_templates(self, unit_id, mode="mean"):
+        if "template" in self._get_unit_property_names():
+            template = self._get_unit_property_values("template")
+            if len(template) > 0:
+                if mode == "mean":
+                    return template.T  # (samples, no_channels)
+                elif mode == "std":
+                    return
+
+    def add_epochs(self):
+        """
+        Auxiliary static method for nwbextractor.
+
+        Adds epochs from recording object to nwbfile object.
+
+        """
+        if self.nwbfile is not None:
+            assert isinstance(self.nwbfile, pynwb.NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+
+        # add/update epochs
+        for epoch_name in self.recording.get_epoch_names():
+            epoch = self.recording.get_epoch_info(epoch_name)
+            if self.nwbfile.epochs is None:
+                self.nwbfile.add_epoch(
+                    start_time=self.recording.frame_to_time(epoch["start_frame"]),
+                    stop_time=self.recording.frame_to_time(epoch["end_frame"] - 1),
+                    tags=epoch_name,
+                )
+            else:
+                if [epoch_name] in self.nwbfile.epochs["tags"][:]:
+                    ind = self.nwbfile.epochs["tags"][:].index([epoch_name])
+                    self.nwbfile.epochs["start_time"].data[ind] = self.recording.frame_to_time(epoch["start_frame"])
+                    self.nwbfile.epochs["stop_time"].data[ind] = self.recording.frame_to_time(epoch["end_frame"])
+                else:
+                    self.nwbfile.add_epoch(
+                        start_time=self.recording.frame_to_time(epoch["start_frame"]),
+                        stop_time=self.recording.frame_to_time(epoch["end_frame"]),
+                        tags=epoch_name,
+                    )
+
+    def add_recording(self, segment_index=0):
+        super().add_recording(segment_index=segment_index)
+        if self._conversion_ops["write_electrical_series"]:
+            self.add_epochs()
+
+
+def create_si013_example(seed):
```

Comment:
Why is this here? Should it be moved to tests?

Reply:
https://github.com/catalystneuro/nwb-conversion-tools/pull/230#discussion_r751235770